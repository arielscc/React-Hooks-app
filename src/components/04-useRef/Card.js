import React from 'react';

export const Card = ({ id, title, body }) => {
  return (
    <div className="w-64 p-6 border shadow-xl bg-white rounded-lg self-center justify-self-center">
      <h1 className="font-bold text-lg uppercase text-center mb-2">{title}</h1>
      <hr />
      <p className="mt-2 font-thin">{body}</p>
    </div>
  );
};
