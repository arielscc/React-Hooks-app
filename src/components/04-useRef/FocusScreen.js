import React, { useRef, useState } from 'react';

export const FocusScreen = () => {
  const initialState = {
    value: '',
    copy: false,
  };
  const [state, setState] = useState(initialState);
  const inputRef = useRef();

  const { value, copy } = state;

  const handleClick = () => {
    const input = inputRef.current;
    input.select();
    document.execCommand('copy');
    setState({
      ...state,
      copy: true,
    });
    setTimeout(() => {
      setState({
        ...state,
        copy: false,
      });
    }, 500);
  };

  const handleChange = (e) => {
    setState({
      ...state,
      value: e.target.value,
    });
  };
  return (
    <div className="w-4/5 m-auto mt-20 text-gray-800">
      <h1 className="text-3xl text-center font-bold">Focus Screen</h1>
      <hr />
      <input
        ref={inputRef}
        value={value}
        onChange={handleChange}
        className=" w-full my-2 rounded-lg border py-2 px-5 shadow-sm"
        placeholder="Nombre"
      />
      <button
        className="rounded-lg bg-indigo-500 text-white font-semibold  py-1 px-6"
        onClick={handleClick}
      >
        Focus
      </button>
      {copy && <p>Copied!</p>}
    </div>
  );
};
