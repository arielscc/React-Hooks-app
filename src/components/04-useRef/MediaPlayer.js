/**
 * @ Este componente presenta un error en la renderizaciÃ³n del video
 *    este no se muestra en pantalla
 **/
import React from 'react';

const MediaPlayer = () => {
  return (
    <div>
      <video width="1000" title="video de planeta" controls>
        <source src="video/planeta.mp4" type="mp4" />
      </video>

      <video src="/video/planeta.mp4" controls></video>

      <button>play</button>
    </div>
  );
};

export default MediaPlayer;
