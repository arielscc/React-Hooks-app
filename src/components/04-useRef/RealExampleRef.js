import React, { useState } from 'react'
import { MultipleCustomHook } from '../03-Examples/MultipleCustomHook'

export const RealExampleRef = () => {
  const [show, setShow] = useState(true);

  return (
    <div className="w-4/5 m-auto mt-20 text-gray-800">
      <h1 className="text-3xl text-center font-bold">Real Example Ref</h1>
      <hr/>
      {
        show && <MultipleCustomHook/>
      }
      <button onClick={ () => setShow( !show ) }>ver</button>
    </div>
  )
}
