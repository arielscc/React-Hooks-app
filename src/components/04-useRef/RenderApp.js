import React, { useEffect, useRef, useState } from 'react';

export const RenderApp = () => {
  const [text, setText] = useState('');
  const renderRef = useRef(1); //

  useEffect(() => {
    renderRef.current += 1;
    return () => {
      // cleanup
    };
  });
  return (
    <div className="p-8">
      <input
        type="text"
        value={text}
        onChange={(e) => setText(e.target.value)}
        className="border-2 border-teal-700"
      />
      <h1>renders: {renderRef.current}</h1>
    </div>
  );
};
