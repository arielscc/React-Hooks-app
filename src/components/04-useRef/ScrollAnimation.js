import React, { useEffect, useRef, useState } from 'react';

export const ScrollAnimation = () => {
  const [background, setBackground] = useState('tomato');
  const divRef = useRef();

  useEffect(() => {
    const handleScroll = () => {
      const div = divRef.current;
      const { y } = div.getBoundingClientRect();
      y > 10 ? setBackground('tomato') : setBackground('teal');
    };
    window.addEventListener('scroll', handleScroll);
    return () => {
      window.removeEventListener('scroll', handleScroll);
    };
  }, []);
  return (
    <div style={{ height: '180vh', background }}>
      <div className="h-24"></div>
      <div ref={divRef}>
        <p>Hola</p>
      </div>
    </div>
  );
};
