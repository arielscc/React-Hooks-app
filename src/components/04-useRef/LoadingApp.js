import React, { useRef, useState } from 'react';
import { useEffect } from 'react';
import { getPost } from '../../helpers/getPost';
import { Card } from './Card';
import { TailSpin } from 'svg-loaders-react';

export const LoadingApp = () => {
  const [mostrar, setMostrar] = useState(false);
  const [posts, setPosts] = useState([]);
  const [loading, setLoading] = useState(true);

  const isMountedRef = useRef(true);

  useEffect(() => {
    updatePost();
    return () => {
      isMountedRef.current = false;
    };
  });

  const updatePost = () => {
    getPost().then((data) => {
      if (isMountedRef.current) {
        setPosts(data);
        setLoading(false);
      }
    });
  };

  return (
    <>
      <div className="flex flex-col">
        <h1 className="w-full text-center my-4 text-2xl font-bold text-gray-700">
          CARDS
        </h1>
        <button
          onClick={() => setMostrar(!mostrar)}
          className="w-32 py-2 px-4 rounded bg-gray-700 text-white font-bold focus:outline-none self-center"
        >
          {mostrar ? 'ocultar' : 'mostrar'}
        </button>
      </div>
      {mostrar && (
        <div
          className="my-24 px-16 grid
        xl:grid-cols-5 lg:grid-cols-4 md:grid-cols-3 sm:grid-cols-2 gap-10 justify-center"
        >
          {loading ? (
            <TailSpin stroke="teal" />
          ) : (
            posts.map((post) => {
              return <Card key={post.id} {...post} />;
            })
          )}
        </div>
      )}
    </>
  );
};
