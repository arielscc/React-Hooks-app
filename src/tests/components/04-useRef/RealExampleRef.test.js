import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';
import '@testing-library/jest-dom';

describe('Pruebas en RealExampleRef', () => {
  test('Debe mostrarse correctamente', () => {
    const wrapper = shallow(<RealExampleRef />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar el componente <MultipleCustomHook/>', () => {
    const wrapper = shallow(<RealExampleRef />);

    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultipleCustomHook').exists()).toBe(true);
  });
});
