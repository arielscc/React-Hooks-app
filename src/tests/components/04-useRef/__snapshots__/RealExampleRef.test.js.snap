// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en RealExampleRef Debe mostrarse correctamente 1`] = `
<div
  className="w-4/5 m-auto mt-20 text-gray-800"
>
  <h1
    className="text-3xl text-center font-bold"
  >
    Real Example Ref
  </h1>
  <hr />
  <button
    onClick={[Function]}
  >
    ver
  </button>
</div>
`;
